mllib/src/main/scala/org/apache/spark/ml/regression/LinearRegression.scala

```diff
@@ -958,14 +973,17 @@ private class LeastSquaresCostFun(
 
   override def calculate(coefficients: BDV[Double]): (Double, BDV[Double]) = {
     val coeffs = Vectors.fromBreeze(coefficients)
+    val bcFeaturesStd = instances.context.broadcast(featuresStd)
+    val bcFeaturesMean = instances.context.broadcast(featuresMean)
+    val bcCoeffs = instances.context.broadcast(coeffs)
```

Comment:
Can we move ```bcFeaturesStd``` and ```bcFeaturesMean``` out of ```calculate```? Since they are invariables during the iterations, we should not broadcast them in each iterations. 
Meanwhile, it's better to call ```destroy(blocking = false)``` to release memory of broadcast variables in the right place and time.